backend/cfg/vectorize.ml

```diff
@@ -134,7 +138,327 @@ module Dependency_graph = struct
       ~terminator:(fun instruction -> print_node (`Terminator instruction))
 end
 
-let dump ppf cfg_with_layout ~(dependency_graph : Dependency_graph.t) ~msg =
+module Adjacent_memory_accesses = struct
+  module Address = struct
+    type op =
+      | Load
+      | Store
+
+    type t =
+      { op : op;
+        memory_chunk : Cmm.memory_chunk;
+        addressing_mode : Arch.addressing_mode;
+        instruction : Instruction.t
+      }
+
+    let create_loads (basic_instruction : Cfg.basic Cfg.instruction) : t option
```

Comment:
We can avoid duplication between `create_loads` and  `create_stores`, just define one function `create : basic instruction -> t`  and filter based on `op` field later.